pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -110,8 +111,8 @@ func (c *CloudProvider) Create(ctx context.Context, nodeRequest *cloudprovider.N
 }
 
 // GetInstanceTypes returns all available InstanceTypes
-func (c *CloudProvider) GetInstanceTypes(ctx context.Context) ([]cloudprovider.InstanceType, error) {
-	return c.instanceTypeProvider.Get(ctx)
+func (c *CloudProvider) GetInstanceTypes(ctx context.Context, requestedInstanceTypes sets.Set) ([]cloudprovider.InstanceType, error) {
```

Comment:
This will break with multiple provisioners. The instanceTypeProvider caches the instance types, so this new param won't take affect if the instance types are cached already or if the cache is empty, the `requestedInstanceTypes` set would cause another provisioner to pickup new instance types that they may not be flexible to.